src/graphics/webgpu/webgpu-bind-group-format.js

```diff
@@ -0,0 +1,141 @@
+/** @typedef {import('../bind-group-format.js').BindGroupFormat} BindGroupFormat */
+/** @typedef {import('./webgpu-graphics-device.js').WebgpuGraphicsDevice} WebgpuGraphicsDevice */
+
+import { WebgpuUtils } from './webgpu-utils.js';
+
+/**
+ * A WebGPU implementation of the BindGroupFormat, which is a wrapper over GPUBindGroupLayout.
+ *
+ * @ignore
+ */
+class WebgpuBindGroupFormat {
+    /**
+     * Unique key, used for caching
+     *
+     * @type {string}
+     */
+    key = '';
+
+    /**
+     * @param {BindGroupFormat} bindGroupFormat -
+     */
+    constructor(bindGroupFormat) {
+
+        /** @type {WebgpuGraphicsDevice} */
+        const device = bindGroupFormat.device;
+
+        /** @type {GPUBindGroupLayoutDescriptor} */
+        const descr = this.createDescriptor(bindGroupFormat);
+
+        /** @type {GPUBindGroupLayout} */
+        this.bindGroupLayout = device.wgpu.createBindGroupLayout(descr);
```

Comment:
any reason `createDescriptor` function doesn't also just do this?

Reply:
I find jsdocs syntax so verbose, that I often avoid it for private function only called within the class.